src/main/java/client/ClientTemp.java

```diff
@@ -15,14 +15,14 @@
 
     private final static Logger logger = LoggerFactory.getLogger(ClientTemp.class);
     private final ClientHandler clientHandler;
-
+    private final Session session;
 
     ClientTemp(String name, String hostname, int port, Codec codec, Viewer viewer) throws Exception {
         InetSocketAddress inetSocketAddress = new InetSocketAddress(hostname, port);
         User user = new User(name);
         SocketChannel socketChannel = SocketChannel.open(inetSocketAddress);
-        Session session = new Session(socketChannel, codec, user);
-        clientHandler = new ClientHandler(session, viewer);
+        this.session = new Session(socketChannel, codec, user);
+        clientHandler = new ClientHandler(user, viewer);
```

Comment:
Why do we create the handler inside the part that is responsible for the session level? Probably, we need to pass the handler itself as a parameter in the constructor